core-1/src/main/java/SpringApp.java

```diff
@@ -7,14 +7,21 @@
 
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
-import parkinglot.Car;
+import org.springframework.stereotype.Component;
+import scanning.parkinglot.ProjectConfig;
 
 public class SpringApp {
 
     public static void main(String[] args) {
         // initiate application context there
-        ApplicationContext context = new AnnotationConfigApplicationContext("parkinglot/beans");
+        ApplicationContext context = new AnnotationConfigApplicationContext(ProjectConfig.class);
```

Comment:
/ 3a. REQUIRED CONDITION - Use component scanning while loading the context

return this line back, it was good